app/screens/post_options/index.js

```diff
@@ -65,12 +77,38 @@ function mapStateToProps(state, ownProps) {
         }
     }
 
+    if (ownProps.channelIsReadOnly) {
+        canFlag = false;
+    }
+
+    if (ownProps.isSystemMessage) {
+        canAddReaction = false;
+        canCopyPermalink = false;
+        canEdit = false;
+        canPin = false;
+    }
+    if (ownProps.hasBeenDeleted) {
+        canDelete = false;
+    }
+
+    if (!ownProps.showAddReaction) {
+        canAddReaction = false;
+    }
+
+    if (ownProps.isSystemMessage && ownProps.managedConfig.copyAndPasteProtection !== 'true' && post.message) {
```

Comment:
this should be `!ownProps.isSystemMessage`

Reply:
Sorry, I need to be more careful!